ui/ducks/app/app.ts

```diff
@@ -307,12 +307,12 @@ export default function reduceApp(
     case actionConstants.MODAL_CLOSE:
       return {
         ...appState,
-        modal: Object.assign(
-          appState.modal,
-          { open: false },
-          { modalState: { name: null, props: {} } },
-          { previousModalState: appState.modal.modalState },
-        ),
+        modal: {
```

Comment:
This change is to prevent Redux from raising a state management error in some cases.

Reply:
These seem really similar to me. How is the before and after code different? How would it fix an error?

Reply:
The previous version was mutating `appState.modal`, but we're not supposed to mutate Redux state. It should be updated immutable, creating new objects rather than mutating.

It seems like a legitimate bug fix to me, though maybe better submitted as a separate PR

Reply:
Exactly, but I agree it would be better to have this fix in a separate PR. Can we make an exception in this case? <3